plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/blobstore/AzureBlobContainer.java

```diff
@@ -72,14 +72,14 @@ public InputStream readBlob(String blobName) throws IOException {
         logger.trace("readBlob({})", blobName);
 
         if (!blobExists(blobName)) {
```

Comment:
I know that this is a part of another PR, but since we are modifying it here - why did we need it in the first place? It looks like we are handling file not found condition when we try to read from the stream anyway. Why do it twice?